cmd/juju/status/status.go

```diff
@@ -408,15 +408,25 @@ func (c *statusCommand) statusArgsWithoutWatchFlag(args []string) []string {
 	var jujuStatusArgsWithoutWatchFlag []string
 
 	for i := range args {
+		// In order to support gnu flags, we must first check if the
+		// watch flag is using gnu style. In that case, we must remove
+		// the entire arg, since it's one entire string (e.g.
+		// --watch=1s).
+		if strings.HasPrefix(args[i], "--watch=") {
+			jujuStatusArgsWithoutWatchFlag = append(args[:i], args[i+1:]...)
+			break
+		}
+		// If the flag is not using gnu style, we must remove both the
+		// flag and the argument (e.g --watch 1s)
 		if args[i] == "--watch" {
 			jujuStatusArgsWithoutWatchFlag = append(args[:i], args[i+2:]...)
-			if !c.noColor {
-				jujuStatusArgsWithoutWatchFlag = append(jujuStatusArgsWithoutWatchFlag, "--color")
-			}
 			break
 		}
 	}
 
+	if !c.noColor {
+		jujuStatusArgsWithoutWatchFlag = append(jujuStatusArgsWithoutWatchFlag, "--color")
+	}
```

Comment:
This is a side effect. I'm guessing the reason we have this here is to ensure we pass `--color` to ensure color is explict for viddy to avoid bugs? The fact it isn't obvious is a smell

Perhaps we should rename the function to `statusCommandForViddy` or something?

Reply:
done